cinema/views.py

```diff
@@ -56,3 +89,66 @@
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+
+        date = self.request.query_params.get("date")
+        movie = self.request.query_params.get("movie")
+
+        if self.action == "list":
+            queryset = (
+                queryset
+                .select_related()
+                .annotate(
+                    tickets_available=(
+                        F("cinema_hall__rows") * F("cinema_hall__seats_in_row")
+                        - Count("tickets")
+                    )
+                )
+            )
+
+        if self.action == "retrieve":
+            queryset = queryset.select_related("cinema_hall", "movie")
+
+        if date:
+            queryset = queryset.filter(show_time__date=date)
+
+        if movie:
+            movie = _params_to_ints(movie)
+            queryset = queryset.filter(movie__id__in=movie)
+
+        return queryset.distinct()
+
+
+class OrderSetPagination(PageNumberPagination):
+    page_size = 3
+    page_size_query_param = "page_size"
+    max_page_size = 20
+
+
+class OrderViewSet(viewsets.ModelViewSet):
+    queryset = Order.objects.all()
+    serializer_class = OrderSerializer
+    pagination_class = OrderSetPagination
+
+    def get_queryset(self):
+        queryset = self.queryset.filter(user=self.request.user)
```

Comment:
In `OrderViewSet`, ensure that the user is authenticated before filtering orders by user. Consider adding authentication checks to prevent unauthorized access.